driver-core/src/test/java/com/datastax/driver/core/AuthenticationTest.java

```diff
@@ -132,4 +137,29 @@ public void run() {
         }, 2000);
     }
 
+    /**
+     * Ensures that the driver throws an AuthenticationException
+     * when an authentication error occurs during connection pool initialization.
+     *
+     * @jira_ticket JAVA-1431
+     */
+    @Test(groups = "short")
+    @CCMConfig(dirtiesContext = true)
+    public void should_not_connect_with_wrong_credentials() throws InterruptedException {
+        PlainTextAuthProvider authProvider = new PlainTextAuthProvider("cassandra", "cassandra");
+        Cluster cluster = Cluster.builder()
+            .addContactPoints(getContactPoints())
+            .withPort(ccm().getBinaryPort())
+            .withAuthProvider(authProvider)
+            .build();
+        cluster.init();
+        authProvider.setPassword("wrong");
```

Comment:
One concern, what if you have 1 node in the cluster which for whatever reason you can't authenticate with?  In this case, you wouldn't be able to establish a Session.

One possibility that I could think of is if for whatever reason a node you are trying to authenticate with can't access the replicas owning data for that user id in the system_auth table and thus can't authenticate the connection.   This seems very unlikely, but in such a case it seems like it could be a bigger problem than the one this ticket attempts to solve.

I think triggering the host down and starting reconnection and not propagating the exception up to the user might be a good happy medium as you can then establish connections to the rest of the nodes (in this case).  wdyt?